python/TestHarness/TestHarness.py

```diff
@@ -371,37 +386,57 @@ def testOutputAndFinish(self, tester, retcode, output, start=0, end=0):
       caveats = test['caveats']
 
     if self.options.pbs and self.options.processingPBS == False:
-      (reason, output) = self.buildPBSBatch(output, tester)
+      result, output = self.buildPBSBatch(output, tester)
+      # Error launching job according to cluster launcher
+      if result != '':
+        # Setting the status bucket in the following way is not ideal, but the
+        # tester never launched accoding to `qstat`. So set the status bucket now,
+        # and allow the Testharness to deal with it, as a failed test accordingly.
+        tester.setStatus(result, 'FAIL')
+
     elif self.options.dry_run:
-      reason = 'DRY_RUN'
-      test['status_message'] = reason
+      # Set the successful message for DRY_RUN
+      tester.success_message = 'DRY RUN'
       output += '\n'.join(tester.processResultsCommand(self.moose_dir, self.options))
     else:
-      (reason, output) = tester.processResults(self.moose_dir, retcode, self.options, output)
+      output = tester.processResults(self.moose_dir, retcode, self.options, output)
 
     if self.options.scaling and test['scale_refine']:
       caveats.append('scaled')
 
-    did_pass = True
-    if reason == '':
-      # It ran OK but is this test set to be skipped on any platform, compiler, so other reason?
+    # If the TestHarness set any statuses (like RUNNING...) and now the test has
+    # finished, check if recode is 0 and place the test in the success bucket.
+    if retcode == 0 and self.options.processingPBS:
+      tester.setStatus(tester.success_message, tester.success_bucket)
+
+    # Check for test failure using the status bucket
+    did_pass = tester.didPass()
+    status = tester.getStatus()
+    result = ''
+
+    # PASS and DRY_RUN fall into this catagory
+    if did_pass:
       if self.options.extra_info:
         checks = ['platform', 'compiler', 'petsc_version', 'mesh_mode', 'method', 'library_mode', 'dtk', 'unique_ids']
         for check in checks:
           if not 'ALL' in test[check]:
             caveats.append(', '.join(test[check]))
       if len(caveats):
-        result = '[' + ', '.join(caveats).upper() + '] ' + tester.getStatusMessage()
-      elif self.options.pbs and self.options.processingPBS == False:
-        result = 'LAUNCHED'
+        result = '[' + ', '.join(caveats).upper() + '] ' + tester.getSuccessMessage()
       else:
-        result = tester.getStatusMessage()
-    elif reason == 'DRY_RUN':
-      result = 'DRY_RUN'
+        result = tester.getSuccessMessage()
+
+    # FAIL, DIFF and DELETED fall into this catagory
+    elif status == 'FAIL' or status == 'DIFF' or status == 'DELETED':
+      result = 'FAILED (%s)' % tester.getStatusMessage()
```

Comment:
Something strange is happening here. I do not believe this code is being hit. The only thing that seems to be being handled by this function, is PASSING tests. Over the course of this PR, I have whittled  down the amount of logic this method performs. I believe anything that is failing, is being handled in 'handleTestResult' method instead.

Unless, and because I have been using moose/test for the majority of testing this PR, none of the testers are truly failing. That everything that I have seen 'FAIL', is stuff we _want_ and _expect_ to fail, and is handled not by the testers, but by the TestHarness. Which does not use this function, but instead sends expected failures directly to 'handleTestResult'.